src/services/character/character.service.ts

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash'
+import { Query, Document } from 'mongoose'
+import characters from '../../data/models/character-schema'
+import { ICharacter } from '../../types/character/character-type'
+
+export class CharacterService {
+  private static _instance: CharacterService
+
+  public static getInstance(): CharacterService {
+    if(_.isNil(CharacterService._instance)) {
+      CharacterService._instance = new CharacterService()
+    }
+    return CharacterService._instance
+  }
+
+  public async getEntity(id: string):Promise<Document | null> {
```

Comment:
Are you sure about null?
I am not.
Test to remove null and see if there is a typing error.
Since you wrap the character inside a if, nothing is returning null.
Also throwing an error in the code should be typed as well with never.
So the minimum here should be Document | never.
If there is an error without null, reach me to talk about it.

Reply:
I will try to change the type of that function.

Reply:
Yes, there is a typing error that a if doesn't remove, sadly enough... And typing with never doesn't work.

Reply:
Well ok let just skip this.